pulp_2to3_migration/app/pre_migration.py

```diff
@@ -211,6 +211,25 @@ def pre_migrate_content_type(content_model, mutable_type, lazy_type, premigrate_
             )
             pulp2content_batch = Pulp2Content.objects.bulk_create(pulp2content,
                                                                   ignore_conflicts=True)
+
+            # bulk_create(ignore_conflicts=True) hands back the same item-set we passed in,
+            # *even if* it decided to update an existing db-record rather than creating a new
+            # one with the passed-in PK. As a result, we can't trust pulp2content_batch to
+            # have the 'right' PKs (i.e., the in-memory p2content_batch doesn't match the
+            # db-reality). This causes the pre_migrate_content_detail() below to fail as it
+            # attempts to create detail-records for the Pulp2Content records it's been handed.
+            # THEREFORE - we need to find the 'real' IDs of everything in p2content-batch based
+            # on its uniqueness-fields and update the in-memory list with them.
+            for p2c in pulp2content_batch:
+                filter_q = Q(
+                    pulp2_content_type_id=content_type,
+                    pulp2_id=p2c.pulp2_id,
+                    pulp2_repo=p2c.pulp2_repo,
+                    pulp2_subid=p2c.pulp2_subid,
```

Comment:
I don't know if this is actually the case, but filtering on a *maybe* unindexed field (unindexed for everything other than errata) might cause perf degredation.

Reply:
pulp_deb uses it as well. Also, this code can't "know" anything about plugins - it only knows that these 4 fields are the 'natural keys' of Pulp2Content, and we can't rely on PK. So, here we are :( 

This whole loop is a performance hit. Unfortunately, the behavior of bulk_create() leaves us no other choice. Slow is bad - but FK-violation-exceptions are way worse.